proposals/extensions.md

```diff
@@ -116,6 +109,251 @@ public implicit extension E for S
 }
 ```
 
+## Open issue: disambiguating between extension type method and classic extension method
+
+Should we disambiguate when signatures match?
+```csharp
+var x = new C().M; // ambiguous
+new C().M(); // ambiguous once we mix all extensions
+
+class C { }
+
+implicit extension E for C
+{
+    public static void M() { }
+}
+
+static class Extensions
+{
+    public static void M(this C c) { }
+}
+```
+
+## Open issue: removing zero-arity-matches-any from var scenario
+
+By the current rules, there are some unfortunate interactions between member access and natural function type.
+> Note: When the result of such a member lookup is a method group and K is zero,
+> the method group can contain methods having type parameters.
+> This allows such methods to be considered for type argument inferencing. end note
+
+I would like to brainstorm tweaks to member access and natural function type to make this work better.
+
+```csharp
+var x = C.Member; // error: member lookup finds C.Member (method group) and lacks type arguments to apply to that match
+
+class C
+{
+    public static void Member<T>() { }
+}
+
+implicit extension E for C
+{
+    public static int Member = 42;
+}
+```
+
+## Open issue: behavior of `static using` directives
+
+We'll need to update "extension member lookup section" and "using static directives" to allow the following two scenarios:
+
+```csharp
+using static C;
+using static D;
+
+Nested1.M(); // Nested1 should be in scope too (as if it were declared as a static member of C)
+Nested2.M();
+
+class C { }
+implicit extension E for C
+{
+    public class Nested1 { public static void M() { }  }
+}
+
+class D { public class Nested2 { public static void M() { } }}
+```
+
+```csharp
+using static Extension;
+
+// M and Nested should be in scope
+
+explicit extension Extension for object
+{
+    public static void M() { }
+    public class Nested { }
+}
+```
+
+## Open issue: what kind of type is an extension type?
+
+Many sections of the spec need to consider the kind of type. Consider a few examples:  
+- the spec for a conditional element access `P?[A]B`
+considers whether `P` is a nullable value type. So it will need to handle the case
+where `P` is an instance of an extension type on a nullable value type,
+- the spec for an object creation considers whether the type is
+a value_type, a type_parameter, a class_type or a struct_type,
+- the spec for satisfying constraints also consider what kind of type were dealing with.
+
+It may be possible to address all those cases without changing each such section of the spec,
+but rather by adding general rules ("an extension on a class type is considered a class type" or some such).
+
+## Open issue: need to specify type erasure
+
+It should be done in a way that allows switching between the extension type and the underlying type without binary break.  
+It should allow for using type parameters as type parameters: `void M<T>(E<T> e)`.  
```

Comment:
Did you mean to say: "It should allow for using type parameters as type **arguments**" ?